python/ray/_private/test_resource_isolation_config.py

```diff
@@ -0,0 +1,218 @@
+import pytest
+
+from ray._private import utils
+from ray._private.resource_isolation_config import ResourceIsolationConfig
+
+
+def test_disabled_by_default():
+    resource_isolation_config = ResourceIsolationConfig()
+    assert not resource_isolation_config.is_enabled()
+
+
+def test_disabled_isolation_with_cgroup_path_raises_exception():
+    with pytest.raises(ValueError):
+        ResourceIsolationConfig(
+            enable_resource_isolation=False, cgroup_path="/some/path"
+        )
+
+
+def test_disabled_isolation_with_reserved_cpu_raises_exception():
+    with pytest.raises(ValueError):
+        ResourceIsolationConfig(enable_resource_isolation=False, system_reserved_cpu=1)
+
+
+def test_disabled_isolation_with_reserved_memory_raises_exception():
+    with pytest.raises(ValueError):
+        ResourceIsolationConfig(
+            enable_resource_isolation=False, system_reserved_memory=1
+        )
+
+
+def test_enabled_invalid_cgroup_path_type():
+    with pytest.raises(ValueError):
+        ResourceIsolationConfig(enable_resource_isolation=True, cgroup_path=1)
+
+
+def test_enabled_invalid_reserved_cpu_type():
+    with pytest.raises(ValueError):
+        ResourceIsolationConfig(enable_resource_isolation=True, system_reserved_cpu="1")
+
+
+def test_enabled_invalid_reserved_memory_type():
+    with pytest.raises(ValueError):
+        ResourceIsolationConfig(enable_resource_isolation=True, system_reserved_cpu="1")
+
+
+def test_enabled_default_config_proportions(monkeypatch):
+    object_store_memory = 10 * 10**9
+    total_system_memory = 128 * 10**9
+    total_system_cpu = 32
+    monkeypatch.setattr(
+        utils, "get_system_memory", lambda *args, **kwargs: total_system_memory
+    )
+    monkeypatch.setattr(utils, "get_num_cpus", lambda *args, **kwargs: total_system_cpu)
+    resource_isolation_config = ResourceIsolationConfig(enable_resource_isolation=True)
+    resource_isolation_config.add_object_store_memory(object_store_memory)
+    # expect the default to be the min(128 * 0.10, 25G) + object_store_memory
+    expected_reserved_memory = 22800000000
+    # expect the default to be the min(32 * 0.05, 1)/32 * 10000
+    expected_reserved_cpu_weight = 312
+    assert resource_isolation_config.system_reserved_memory == expected_reserved_memory
+    assert (
+        resource_isolation_config.system_reserved_cpu_weight
+        == expected_reserved_cpu_weight
+    )
+
+
+def test_enabled_default_config_values(monkeypatch):
+    object_store_memory = 10 * 10**9
```

Comment:
Just FYI (no change requested), I found we have `humanfriendly` package already checked in.
```sh
[~/ray] (hjiang/compilation-options) [gke_hjiang-proj-cgroup-experiment_us-west1_hjiang-cgroup-test]
ubuntu@hjiang-devbox-pg$ git grep "humanfriendly"
doc/source/ray-overview/pip_freeze_ray-ml-py39-cpu.txt:humanfriendly==10.0
python/requirements_compiled.txt:humanfriendly==10.0
```
In theory we should be able to do
```py
object_store_memory = humanfriendly.parse_size("10GB")
```